Update header

jameseaster opened the pull request:
#### _Resolves [574](https://github.com/digitalbazaar/veres-wallet/issues/574) from veres-wallet_

_____

### What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- UI update header and drawer
- Fix vue linting errors

<br/>

### What is the current behavior? (You can also link to an open issue here)
- The "Settings" and "Log Out" options are inside of the avatar in the top right of the screen and not in the app drawer as options
- https://github.com/digitalbazaar/veres-wallet/issues/574

<br/>

### What is the new behavior?
- The person avatar has been removed and the "Settings" and "Log Out" are available on the app header & app drawer

<br/>

### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
- n/a

<br/>

### Screenshots:
> Updated app header
![header](https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/48fc531d-bcdc-4fb1-aa45-f773a9c0eda3)

> Updated app drawer
![drawer](https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/9d1811ba-f2a9-41ee-88ab-dce336584c4f)

> Items accessible in drawer via hamburger menu on small screen
<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/62577b5c-cb37-4073-9676-386e834d6441" alt="mobile" height="550"/>

<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/2d57973d-85bd-4453-9f92-3c35f14088bf" alt="drawer" height="550"/>

aljones15 reviewed (approved):
LGTM, but do await other approval. There are some vue lint errors that should be address sometime, but I don't think are related to this PR. P.S. while I did approve I'm not sure I approve of hard-coding something to get around a vue warning. We should probably look into a better solution for the warning or run your local env with `NODE_ENV=development`

aljones15 commented on lib/index.js:
```diff
@@ -17,6 +15,10 @@ export {routeGuards} from './router.js';
 // load bedrock-vue-wallet config
 import './config.js';
 
+// Satisfies vue console warning
```
What is the vue warning this negates?

jameseaster replied:
Vue generates this console warning when those values are not specified

![warnings](https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/a224272a-ae02-42f5-8909-618f8f103c69)

aljones15 replied:
We might want the PROD_DEVTOOLS are some point, I get the feeling this is supposed to be set somewhere else too.

jameseaster replied:
I removed this change per @dlongley 's comment at the bottom. I will create an issue to address this so that we can more appropriately address this vue warning in the future. See [481125d](https://github.com/digitalbazaar/bedrock-vue-wallet/pull/62/commits/481125d4e3285f5a662da99c1e5832067be4efb1)

jameseaster replied:
_New Issue: #64_

aljones15 commented on routes/SettingsPage.vue:
```diff
@@ -169,7 +169,7 @@ export default {
 
     const settingsPageMode = computed(() => {
-      return route.name === 'settings-profile' ?
+      return route && route.name === 'settings-profile' ?
```
you can use `route?.name` if you want here.

davidlehn commented on CHANGELOG.md:
```diff
@@ -1,5 +1,19 @@
 # bedrock-vue-wallet ChangeLog
 
+## 21.2.1 - 2023-10-05
```
Use a temporary date during development.  We update this when doing a release.
```suggestion
## 21.2.1 - 2023-10-xx
```

dlongley commented on CHANGELOG.md:
```diff
@@ -1,5 +1,19 @@
+## 21.2.1 - 2023-10-xx
+
+### Added
+- Add "Settings" and "Log Out" options to header & drawer.
```
This looks good and is an improvement on wide screens. What happens on narrow screens?

jameseaster replied:
(Just added this to the PR content above as well)

Items get moved over to the app drawer with the other navigation links

<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/c49f71b9-f387-4202-b991-75abfe7ec938" width="275" />

<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/6456c1d9-4317-43cd-b670-fdffb3dff7b8" width="275" />

dlongley commented on CHANGELOG.md:
```diff
@@ -1,5 +1,19 @@
+- Add "Settings" and "Log Out" options to header & drawer.
+
+### Changed
+- Move establishing icon library to vue component in order to run code locally with npm link.
```
This is a breaking change. Any application that uses this library but doesn't use the `WalletLayout` component will need to set the default icon set now, right? Do we need to do this?

Will it mean that every component needs to import / set the icon library? If not, which ones should do it and how should it be decided? Will this be an easy decision for developers? Is this new approach good ... bad? Pros/cons?

Couldn't the code already be run locally without making this change?

jameseaster replied:
I was unable to run the code locally without this change. This is the error from the console that prevents the app from loading. For some reason the _Quasar_ variable is undefined in this function and is unable to set the iconSet. So I looked for alternative ways to set the icons. Seeing that WalletLayout utilizes the q-layout component from Quasar which is ["...designed to manage the entire window and wrap page content..."](https://quasar.dev/layout/layout/), I thought it would be the best solution to establish the icon set. 

**I could very easily not be seeing the whole picture**, but I didn't expect a use case for an application to want to use this library without the WalletLayout component since this is a UI library made to generate a wallet UI?

This is the console error if you run the code prior to the changes in this PR:
![Screenshot 2023-10-06 at 10 38 26 AM](https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/183d0f75-0187-4733-ad48-d36875b4bbb8)

This is the line of code in bedrock-vue-wallet > lib > index.js (line 58 before changes from this PR)
![Screenshot 2023-10-06 at 10 40 05 AM](https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/4e24cb40-436b-49a1-8f31-426d936999ae)

If that line is commented out the app will render, but without several icons:
<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/3ccab180-db01-495e-b99a-699cf16e0c47" height="400" />

<img src="https://github.com/digitalbazaar/bedrock-vue-wallet/assets/56396286/0c779644-7f4e-4c47-af5e-10f62a7fd0fc" height="400" />

Would value any and all feedback and perspectives here!

jameseaster replied:
The breaking change notated at the top of this comment has been included in the CHANGELOG.md b9972b0

jameseaster replied:
I reverted the change made to iconSet and updated the change log so there should be no breaking changes with this PR 2e4fd42

dlongley reviewed (changes requested):
Thanks! This is definitely an improvement on wide screens. Can we get screen shots on narrow screens, i.e., smallest supported widths for phone screens?

dlongley commented:
Other notes:

If we're getting Vue warnings, we should figure out the right place to address those properly -- it may be this package or it may be something that is expected in top-level applications. I see there are some Vue linting errors here -- and recognize that they are probably not the result of any changes made in this PR. If we could address them easily now that's great, but it's ok if not -- and we should file an issue to get them cleaned up. Similarly, we could file an issue to figure out what to do about the Vue devtools warnings before trying to do something about them in this PR that may / may not be the right course.

jameseaster commented:
@dlongley Great [point](https://github.com/digitalbazaar/bedrock-vue-wallet/pull/62#issuecomment-1749510954). I removed the quick fix for the Vue feature flags warning and created a new issue to more adequately address it (#64). I also addressed the other vue linting warnings as well  f13a64f

dlongley reviewed (approved):
LGTM, AFAICT. Thanks!
